src/renderer/features/proxies/workers/proxy-worker.ts

```diff
@@ -74,11 +74,13 @@ function initConnection(chain?: Chain, connection?: Connection) {
         return;
       }
 
-      provider.on('connected', async () => {
-        state.apis[chain.chainId] = await ApiPromise.create({ provider, throwOnConnect: true, throwOnUnknown: true });
+      provider.connect().then(() => {
+        provider!.on('connected', async () => {
+          state.apis[chain.chainId] = await ApiPromise.create({ provider, throwOnConnect: true, throwOnUnknown: true });
 
-        console.log('proxy-worker: provider connected successfully');
-        resolve(InitConnectionsResult.SUCCESS);
+          console.log('proxy-worker: provider connected successfully');
```

Comment:
I was discussing about the logs being quite "spammy" in the console with Yaroslav. Any chance we can remove this? Or do you think it's too valuable?

Reply:
I've raised this question in our dev chat. Going to create a task (it might appear in gh issues)